lib/libc/stdlib/malloc/mrs/mrs.c

```diff
@@ -1257,7 +1257,7 @@ static void *mrs_realloc(void *ptr, size_t size) {
 
 	if (size == 0) {
 		mrs_free(ptr);
-		return NULL;
+		ptr = NULL;
```

Comment:
I actually think we might want a different fix for this (see my last fixup in #1891).  Namely, if realloc() returns NULL, that indicates an error and that the original pointer has _not_ been free'd.  If for some reason mrs_malloc(0) returned NULL, we need to not free the old object, so my fixup commit there just removes the 'size == 0' case entirely.